### PR TITLE
general bug fixed

### PR DESCRIPTION
**Pull request by mhijaziB**
- gem shop not instantiating gems fixed: type error
- GemStoreController: cached strings to prevent string typos
- Added null checks for hats and character items gameobject
- Removed duplicate calls for the matchmaking service

### CODE REVIEW

**Review by Leinnan — APPROVED** [review]
Nice one! LGTM

File: client/Assets/Scripts/Game/MatchmakingBehaviour.cs [hunk 1]
```diff
@@ -42,43 +42,43 @@ public async void FindGame()
 			await _beamContext.OnReady;
 
 			Debug.Log($"Starting matchmaking with game_type={GameTypeRef.Id} override timeout={_configuration.OverrideMaxMatchmakingTimeout} ...");
-			var handle = await _beamContext.Api.Experimental.MatchmakingService.StartMatchmaking(GameTypeRef.Id);
+			// var handle = await _beamContext.Api.Experimental.MatchmakingService.StartMatchmaking(GameTypeRef.Id);
```
Comment (allister-beamable) [inline client/Assets/Scripts/Game/MatchmakingBehaviour.cs:45]: Let's avoid leaving commented-out lines in the code unless they are needed for illustrative purposes (and have lots of comments explaining them 😺). In this case, it sounds like the double call to StartMatchmaking was a leftover mistake.

I'd say open another small PR to delete this commented-out line.

File: client/Assets/Scripts/Game/UI/GemStoreController.cs [hunk 2]
```diff
@@ -54,7 +58,7 @@ async void Start()
     // Update is called once per frame
     void Update()
     {
-        var desiredY = IsOpen ? 0 : -Screen.height*2;
+        var desiredY = IsOpen ? 0 : -Screen.height * 3f;
```
Comment (allister-beamable) [inline client/Assets/Scripts/Game/UI/GemStoreController.cs:61]: Might be worth making this a local constant with a descriptive name. As it is, it isn't clear to me why we're multiplying screen height by a scalar. Also, have you tested the visual behavior of this on device? Probably totally fine as it is, but this is the sort of thing that can be subtly weird on different devices 😄

File: client/Packages/packages-lock.json [hunk 3]
```diff
@@ -123,7 +120,7 @@
       "url": "https://packages.unity.com"
     },
     "com.unity.nuget.newtonsoft-json": {
-      "version": "3.2.1",
+      "version": "3.0.2",
```
Comment (allister-beamable) [inline client/Packages/packages-lock.json:123]: Do we know for sure that 3.0.2 is a good version of Newtonsoft to use? What was the reason for downgrading from 3.2.1? If it just sort of happened automatically _AND_ it isn't required transitively by other legitimate dependencies, I would prefer to err on the side of keeping things the same.

**Review by allister-beamable — COMMENTED** [review]
Small after-the-fact review. None of my feedback is big enough to really require a PR, but if you have a chance to apply a little bit of cleanup, I think it would be a net positive for code hygiene ☀️

**Pull request merged**